src/main/java/hudson/security/DangerousMatrixPermissionsAdministrativeMonitor.java

```diff
@@ -48,7 +48,8 @@
 public class DangerousMatrixPermissionsAdministrativeMonitor extends AdministrativeMonitor {
     @Override
     public boolean isActivated() {
-        return !GlobalMatrixAuthorizationStrategy.ENABLE_DANGEROUS_PERMISSIONS && !getSidsWithDangerousPermissions().isEmpty();
+        return ((!GlobalMatrixAuthorizationStrategy.ENABLE_DANGEROUS_PERMISSIONS && !getSidsWithDangerousPermissions().isEmpty()) ||
```

Comment:
`getSidsWithDangerousPermissions().isEmpty()` seems quite expensive to check on every request?

Reply:
That is how the monitor was originally implemented....

`return !GlobalMatrixAuthorizationStrategy.ENABLE_DANGEROUS_PERMISSIONS && !getSidsWithDangerousPermissions().isEmpty(); `

I assume it was originally implemented this way to inform users who had previously used these permissions before they became disabled by default.  Does it make more sense to just show it to users who are still using these permissions now?